modules/core/src/views/map-view.ts

```diff
@@ -27,8 +27,8 @@ export type MapViewState = {
   minPitch?: number;
   /** Max pitch, default `60` */
   maxPitch?: number;
-  /** Viewport center offsets from lng, lat in meters */
-  position?: number[];
+  /** Viewport center offsets from lng, lat, and elevation in meters */
+  position?: [number, number, number];
```

Comment:
This is a stricter type which while more prescriptive can cause breakages in existing applications that are supplying less strictly typed parameters. I suppose that could be a documented change in deck.gl v9.1

Reply:
Approved conditionally, I think there needs to be a mention in the upgrade-guide.md

Reply:
Mention made: https://github.com/visgl/deck.gl/pull/9313/commits/e6b304d31e7d88c19946b65e93ed898e20ee8b3d

Reply:
I share the same concern @ibgreen has but after some research I think this is the right move.. and we could actually make a similar improvement to the other viewports.

- The base `Viewport` position is set to `[0,0,0]` if its missing.
  - [ ] Apply doc and type improvement.
- [`FirstPersonViewport`](https://github.com/visgl/deck.gl/blob/f36b2ffd8a05569f0804d2c57d95402428534a7e/modules/core/src/viewports/first-person-viewport.ts#L25) already types `position` as `[number, number, number]`.
  - [ ] Explicitly state `[x,y,z]` in doc?
- `MapView`'s [projection](https://github.com/visgl/math.gl/blob/f779fc431a4086867f2f43c26e3b0ef2f89da815/modules/web-mercator/src/web-mercator-utils.ts#L315) assumes its length is 3.
- Orbit and Orthographic map `target` to `position`.. the only oddity is otho also supports a [2D offset](https://github.com/visgl/deck.gl/blob/f36b2ffd8a05569f0804d2c57d95402428534a7e/modules/core/src/viewports/orthographic-viewport.ts#L63).
- `GlobeViewport` includes `position` in its type and it relying on `Viewport`'s default implementation.
  - [ ] Apply doc and type improvement.